src/coverage_reporter/openssl_version.cr

```diff
@@ -0,0 +1,30 @@
+module CoverageReporter
+  class OpenSSLVersion
+    WORKS = SemanticVersion.new(1, 1, 0)
+
+    def can_fail?
+      return current.not_nil! < WORKS unless current.nil?
+
+      false
+    end
+
+    private def current
+      matches = /.*?(\d+)\.(\d+)\.(\d+).*/.match(current_string)
+      return nil if matches.nil?
+
+      major = matches[1].to_i
+      minor = matches[2].to_i
+      patch = matches[3].to_i
+      SemanticVersion.new(major, minor, patch)
+    end
+
+    # TODO: rescue from unknown command
+    # TODO: return nil for windows
+    private def current_string
+      # examples:
+      #   OpenSSL 1.0.2k-fips  26 Jan 2017
+      #   OpenSSL 3.0.8 7 Feb 2023 (Library: OpenSSL 3.0.8 7 Feb 2023)
+      `openssl version -v`
```

Comment:
This wouldn't work on default CentOS 7 system, because it only has `openssl-libs` package installed, but the CLI util comes in a separate `openssl` package.

```sh-session
$ docker run -t --rm centos:7 openssl version -v
docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "openssl": executable file not found in $PATH: unknown.

$ docker run -t --rm centos:7 rpm -q 'openssl' 'openssl-libs'
package openssl is not installed
openssl-libs-1.0.2k-19.el7.x86_64
```

Reply:
Thank you! Yes! That's much safer and easier 🔥